src/frontend/src/pages/GanttPage/GanttChartPage.tsx

```diff
@@ -28,176 +29,123 @@ import { SearchBar } from '../../components/SearchBar';
   >([]);
+  const [searchText, setSearchText] = useState<string>('');
 
   /******************** Filters ***************************/
-  const car = query.getAll('car');
-  const showCar1 = car.includes('Car 1');
-  const showCar2 = car.includes('Car 2');
+  const showCars = query.getAll('car').map((car) => parseInt(car));
 
-  const teamCategory = query.getAll('teamCategory');
-  const showElectricalTeamCategory = teamCategory.includes('Electrical');
-  const showMechanicalTeamCategory = teamCategory.includes('Mechanical');
-  const showSoftwareTeamCategory = teamCategory.includes('Software');
-  const showBusinessTeamCategory = teamCategory.includes('Business');
+  const showTeamTypes = query.getAll('teamType');
 
-  const team = query.getAll('team');
-  const showErgonomicsTeam = team.includes('Ergonomics');
-  const showLowVoltageTeam = team.includes('Low Voltage');
-  const showTractiveTeam = team.includes('Tractive');
-  const showDataAndControlsTeam = team.includes('Data and Controls');
-  const showSoftwareTeam = team.includes('Software');
+  const showTeams = query.getAll('team');
 
   const showOnlyOverdue = query.get('overdue') ? query.get('overdue') === 'true' : false;
 
   const expanded = query.get('expanded') ? query.get('expanded') === 'true' : false;
 
   const defaultGanttFilters: GanttFilters = {
-    showCar1,
-    showCar2,
-    showElectricalTeamCategory,
-    showMechanicalTeamCategory,
-    showSoftwareTeamCategory,
-    showBusinessTeamCategory,
-    showErgonomicsTeam,
-    showLowVoltageTeam,
-    showTractiveTeam,
-    showDataAndControlsTeam,
-    showSoftwareTeam,
+    showCars,
+    showTeamTypes,
+    showTeams,
     showOnlyOverdue,
     expanded
   };
 
-  useEffect(() => {
-    if (!projects) return;
-
-    setTeamList(Array.from(new Set(projects.map(getProjectTeamsName))));
-
-    const ganttFilters: GanttFilters = {
-      showCar1,
-      showCar2,
-      showElectricalTeamCategory,
-      showMechanicalTeamCategory,
-      showSoftwareTeamCategory,
-      showBusinessTeamCategory,
-      showErgonomicsTeam,
-      showLowVoltageTeam,
-      showTractiveTeam,
-      showDataAndControlsTeam,
-      showSoftwareTeam,
-      showOnlyOverdue,
-      expanded
+  const filteredProjects = filterGanttProjects(projects ?? [], defaultGanttFilters, searchText);
+  const sortedProjects = filteredProjects.sort(
+    (a, b) => (a.startDate || new Date()).getTime() - (b.startDate || new Date()).getTime()
+  );
+  const ganttTasks = sortedProjects.flatMap((project) => transformProjectToGanttTask(project, expanded));
+
+  if (projectsIsLoading || teamTypesIsLoading || teamsIsLoading || !teams || !projects || !teamTypes)
+    return <LoadingIndicator />;
+  if (projectsIsError) return <ErrorPage message={projectsError.message} />;
+  if (teamTypesIsError) return <ErrorPage message={teamTypesError.message} />;
+  if (teamsIsError) return <ErrorPage message={teamsError.message} />;
+
+  const carHandlerFn = (car: number) => {
```

Comment:
can you rename all these to *FilterHandler instead of *HandlerFn, e.g. carFilterHandler

Reply:
actually nvm, I see why u did it this way now that I read thru the code